packages/frontend/STYLE_GUIDE.md

```diff
@@ -210,6 +214,26 @@ Before moving styles to CSS modules, analyze if they're actually needed:
 <Box p="md" mt="lg">
 ```
 
+#### Text
+
+Avoid using <Text> purely as a generic wrapper for applying styles.
+If you need specific styles (spacing, colors, etc.) that aren’t just standard Mantine typography, use <Box> with component="..." instead.
+This ensures consistency, keeps style props working, and aligns with Mantine’s recommended patterns.
+
+Where possible move existing `<Text>` elements to use nothing, or `<Box>`
+
+```tsx
+// ❌ Bad - Text just for styles
+<Text fw={500} c="dimmed" fz="sm" mt="sm">
+  Parameters
+</Text>
+
+// ✅ Good - inherits local text styles
+<Box component="span" fw={500} c="dimmed" fz="sm" mt="sm">
+  Parameters
+</Box>
```

Comment:
i'm not too sure on this use-case. Wouldn't this cause issues with text elements not being a `p` tag anymore? 🤔

Reply:
yea "aggressive" is the right word here 👍 agree with your points but just thinking about edge cases not covered 🤔